aten/src/ATen/DeviceAccelerator.cpp

```diff
@@ -6,7 +6,7 @@ namespace at::accelerator {
 
 std::optional<c10::DeviceType> getAccelerator(bool checked) {
 #define DETECT_AND_ASSIGN_ACCELERATOR(device_name) \
-  if (at::has##device_name()) {                    \
+  if (at::globalContext().has##device_name()) {    \
```

Comment:
Soooo from a quick chat with @egienvalue looks like there exists some internal config that have CUDA + MTIA in a single binary.
While we fix that on Meta's side, would you be able to comment out the ` DETECT_AND_ASSIGN_ACCELERATOR(MTIA)` so we can land this PR without breaking internal.
I'll work with @egienvalue on the appropriate fix in a follow up PR.

Reply:
I got it. By the way, this fix will introduce an issue tracked by https://github.com/pytorch/pytorch/issues/144567. This issue will be fixed in the subsequent PR https://github.com/pytorch/pytorch/pull/144664.